code/game/turfs/simulated/walls.dm

```diff
@@ -255,20 +255,21 @@
 	//Deconstruction
 	if(iswelder(W))
 		var/obj/item/tool/weldingtool/WT = W
-		if(WT.remove_fuel(1, user))
-			if(engraving)
+		if(engraving)
+			if(WT.remove_fuel(1, user))
 				to_chat(user, "<span class='notice'>You deform the wall back into its original shape")
 				engraving = null
 				engraving_quality = null
 				playsound(src, 'sound/items/Welder.ogg', 100, 1)
 				overlays.Cut()
 				return
+		if(WT.get_fuel() && WT.isOn())
```

Comment:
get_fuel needs to be >=1, not just not-zero, otherwise this will create weird behaviors where it tries to start welding but fails at the do_weld step